`createdir` stops to work

```
./zbox createdir --dirname "/new_folder" --allocation "8fadceafe4d6220ca190af4d40b1407392818269247ef4ca6f58f91f158c54d0"
0chain-core-sdk [INFO]   2022/08/12 17:19:15.544645 wallet.go:290: ******* Wallet SDK Version:v1.7.2-2-g3237f5c-dirty ******* (SetLogFile)
0box-sdk        [INFO]   2022/08/12 17:19:15.544769 sdk.go:73: ******* Storage SDK Version: v1.7.2-2-g3237f5c-dirty *******
0chain-core-sdk [INFO]   2022/08/12 17:19:16.444311 wallet.go:413: ******* Wallet SDK Version:v1.7.2-2-g3237f5c-dirty ******* (InitZCNSDK)
0box-sdk        [DEBUG]  2022/08/12 17:19:17.269600 networkworker.go:105: Get network result:{"miners":["https://demo.0chain.net/miner02","https://demo.0chain.net/miner01","https://demo.0chain.net/miner03"],"sharders":["https://demo.0chain.net/sharder02","https://demo.0chain.net/sharder01"]}
0box-sdk        [INFO]   2022/08/12 17:19:17.862019 dirworker.go:45: Start creating dir for blobbers
0box-sdk        [INFO]   2022/08/12 17:19:18.089465 dirworker.go:146: createdir resp:{"filename":"/new_folder","size":0,"content_hash":"","merkle_root":"","upload_length":0,"upload_offset":0}
0box-sdk        [INFO]   2022/08/12 17:19:18.089733 dirworker.go:151: https://demo.0chain.net/blobber04 /new_folder created.
0box-sdk        [INFO]   2022/08/12 17:19:18.092505 dirworker.go:146: createdir resp:{"filename":"/new_folder","size":0,"content_hash":"","merkle_root":"","upload_length":0,"upload_offset":0}
0box-sdk        [INFO]   2022/08/12 17:19:18.092556 dirworker.go:151: https://demo.0chain.net/blobber05 /new_folder created.
0box-sdk        [INFO]   2022/08/12 17:19:18.304308 dirworker.go:146: createdir resp:{"filename":"/new_folder","size":0,"content_hash":"","merkle_root":"","upload_length":0,"upload_offset":0}
0box-sdk        [INFO]   2022/08/12 17:19:18.304400 dirworker.go:151: https://demo.0chain.net/blobber02 /new_folder created.
0box-sdk        [INFO]   2022/08/12 17:19:18.304361 dirworker.go:146: createdir resp:{"filename":"/new_folder","size":0,"content_hash":"","merkle_root":"","upload_length":0,"upload_offset":0}
0box-sdk        [INFO]   2022/08/12 17:19:18.304433 dirworker.go:151: https://demo.0chain.net/blobber01 /new_folder created.
0box-sdk        [INFO]   2022/08/12 17:19:19.852354 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 17:19:19.852253 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 17:19:19.852297 commitworker.go:102: received a commit request
0box-sdk        [INFO]   2022/08/12 17:19:19.852382 commitworker.go:102: received a commit request
panic: interface conversion: fileref.RefEntity is *fileref.FileRef, not *fileref.Ref

goroutine 55 [running]:
github.com/0chain/gosdk/zboxcore/allocationchange.(*DirCreateChange).ProcessChange(0x103678940?, 0x14000284000)
	/Users/rj/go/pkg/mod/github.com/0chain/gosdk@v1.8.6-0.20220727101848-c573a174d64b/zboxcore/allocationchange/createdir.go:24 +0x2d0
github.com/0chain/gosdk/zboxcore/sdk.(*CommitRequest).processCommit(0x1400069e6c0)
	/Users/rj/go/pkg/mod/github.com/0chain/gosdk@v1.8.6-0.20220727101848-c573a174d64b/zboxcore/sdk/commitworker.go:183 +0x62c
github.com/0chain/gosdk/zboxcore/sdk.startCommitWorker(0x0?, {0x140002ea6c0, 0x40})
	/Users/rj/go/pkg/mod/github.com/0chain/gosdk@v1.8.6-0.20220727101848-c573a174d64b/zboxcore/sdk/commitworker.go:94 +0x3c
created by github.com/0chain/gosdk/zboxcore/sdk.InitCommitWorker
	/Users/rj/go/pkg/mod/github.com/0chain/gosdk@v1.8.6-0.20220727101848-c573a174d64b/zboxcore/sdk/commitworker.go:82 +0x298
```

--- a/zboxcore/allocationchange/createdir.go
+++ b/zboxcore/allocationchange/createdir.go
@@ -21,7 +21,14 @@ func (d *DirCreateChange) ProcessChange(rootRef *fileref.Ref) error {
 	for i := 0; i < len(fields); i++ {
 		found := false
 		for _, child := range dirRef.Children {
-			ref := child.(*fileref.Ref)
+			ref, ok := child.(*fileref.Ref)
+			if !ok {
+				fr, ok := child.(*fileref.FileRef)
+				if ok {
+					ref = &fr.Ref
+				}
+			}
+
 			if ref.Name == fields[i] {
 				dirRef = ref
 				found = true

--- a/zboxcore/allocationchange/createdir.go
+++ b/zboxcore/allocationchange/createdir.go
@@ -1,6 +1,7 @@
 package allocationchange
 
 import (
+	"errors"
 	"path/filepath"
 	"strings"
 
@@ -24,9 +25,10 @@ func (d *DirCreateChange) ProcessChange(rootRef *fileref.Ref) error {
 			ref, ok := child.(*fileref.Ref)
 			if !ok {
 				fr, ok := child.(*fileref.FileRef)
-				if ok {
-					ref = &fr.Ref
+				if !ok {
+					return errors.New("invalid_ref: child node is not valid *fileref.Ref or *fileref.FileRef ")
 				}
+				ref = &fr.Ref
 			}
 
 			if ref.Name == fields[i] {

